### PR TITLE
Week_6_Exercise

### PR DESCRIPTION
**Pull request by CosmusMutuku**
**Exercise Description:**
In this exercise, I:
- explored price prediction for Amazon product data using the Claude model (Anthropic) via multi-shot prompting.
- tested different prompt designs and numbers of examples (0, 2, 5, 7, 10, 30, 50) to evaluate the impact on prediction accuracy.
- The goal was to beat the accuracy of the base model using prompt engineering and few-shot examples.

Findings / Results:
- Multi-shot prompting with 10 examples achieved the best performance: MAE ≈ 16.27, RMSE ≈ 38.59, and 90% of predictions within $20. (Which attained the goal, but on a very basic level)
- Increasing the number of examples beyond 10 (30 or 50) worsened performance, possibly due to variance and overloading the prompt.
- Further slight fine-tuning the model on 10 examples did not outperform prompt engineering (will keep finetuning this further with time and more resources).

Challenges:

- Limited access to Anthropic API; ran out of pay balance and could not add a new payment method (i didn't find a way) to make a purchase.
- No OpenAI keys available to compare models or explore alternative LLMs.

Screenshot: Unfortunately, when I attempted further fine-tuning, I had to restart and rerun the notebook after my Anthropic API key had been disabled. Thus, I was unable to capture screenshots of the results.

### CODE REVIEW

**Comment by ed-donner** [comment]
Ahh very nice - thanks so much for those experiments
Ed
